cmd/zscrub/zscrub.c

```diff
@@ -0,0 +1,242 @@
+/* Scrub a pool from userland */
+
+#include <stdio.h>
+#include <sys/zfs_context.h>
+#include <sys/spa.h>
+#include <sys/dmu.h>
+#include <sys/txg.h>
+#include <sys/dbuf.h>
+#include <sys/zap.h>
+#include <sys/dmu_objset.h>
+#include <sys/spa_impl.h>
+#include <sys/dsl_scan.h>
+#include <libnvpair.h>
+#include <libzfs.h>
+
+libzfs_handle_t *libzfs_handle;
+
+extern int reference_tracking_enable;
+extern int zfs_no_scrub_prefetch;
+extern enum ddt_class zfs_scrub_ddt_class_max;
+
+int verbose = 0;
+
+static void
+print_vdev(char *name, nvlist_t *nv, int depth)
+{
+    nvlist_t **child;
+    uint_t c, children;
+
+    vdev_stat_t *vs;
+    char *state;
+
+    if (nvlist_lookup_nvlist_array(nv, ZPOOL_CONFIG_CHILDREN,
+        &child, &children) != 0)
+        children = 0;
+
+    VERIFY3U(nvlist_lookup_uint64_array(nv, ZPOOL_CONFIG_VDEV_STATS,
+              (uint64_t **)&vs, &c), ==, 0);
```

Comment:
This code doesn't appear to comply with [the sunos style guide](http://www.cis.upenn.edu/%7Elee/06cse480/data/cstyle.ms.pdf), which ZoL [also uses](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#style-guides).